src/chains-config/metadata-consts/index.ts

```diff
@@ -0,0 +1,45 @@
+import { BlockWeightDefinitions, Definition } from '../../types/chains-config';
+import { kusamaDefinitions } from './kusama-consts';
+import { polkadotDefinitions } from './polkadot-consts';
+
+/**
+ * Creates an object that orders each runtime to their appropriate data
+ *
+ * @param definitions An array of objects that group data based on runtimes
+ * and their extrinsicBaseWeight metadata
+ */
+const generateBlockWeightObject = (
+	definitions: Definition[]
+): Record<string, BlockWeightDefinitions> => {
+	const blockWeightObject = {};
+
+	for (let i = 0; i < definitions.length; i++) {
+		const runtimes: number[] = definitions[i].runtimes;
+
+		for (let p = 0; p < runtimes.length; p++) {
+			const version: number = runtimes[p];
+			blockWeightObject[version] = definitions[i].extrinsicBaseWeight;
+		}
+	}
+
+	return blockWeightObject;
+};
+
+/**
+ * Returns a set of runtime pointing to their blockWeightDefinitions specific to
+ * a chain
+ *
+ * @param specName specName from the metadata of the current block being fetched
+ */
+export const getBlockWeight = (
+	specName: string
+): Record<string, BlockWeightDefinitions> | Object => {
+	switch (specName) {
+		case 'polkadot':
+			return Object.freeze(generateBlockWeightObject(polkadotDefinitions));
```

Comment:
Just curious, why do `Object.freeze`?

Reply:
It actually has no purpose at all, removing it :), started off with me testing it against changing types was all.